core/iwasm/interpreter/wasm_runtime.h

```diff
@@ -212,6 +212,7 @@ typedef struct CApiFuncImport {
 
 /* The common part of WASMModuleInstanceExtra and AOTModuleInstanceExtra */
 typedef struct WASMModuleInstanceExtraCommon {
+    void *contexts[WASM_MAX_INSTANCE_CONTEXTS];
```

Comment:
Does it mean that the wasi_ctx always occupies the first slot with key g_wasi_context_key, and user's contexts occupy the following?

Reply:
if wasi is enabled, probably.